src/Nulls.jl

```diff
@@ -150,6 +150,25 @@ julia> coalesce.([null, 1, null], [0, 10, 5])
 coalesce(x) = x
 coalesce(x, y...) = ifelse(x !== null, x, coalesce(y...))
 
+"""
+    levels(x)
+
+Return a vector of unique values which occur or could occur in collection `x`,
+omitting `null` even if present. Values are returned in the preferred order
+for the collection, with the result of [`sort`](@ref) as a default.
+
+Contrary to [`unique`](@ref), this function may return values which do not
+actually occur in the data, and does not preserve their order of appearance in `x`.
+"""
+function levels(x)
+    T = Nulls.T(eltype(x))
+    levs = convert(AbstractArray{T}, filter(!isnull, unique(x)))
```

Comment:
`unique` should never alias, so this might be more efficient with `filter!`.

Reply:
I would have sweared I had used `filter!`. Too many `!` in a row...